network.go

```diff
@@ -2090,20 +2160,23 @@ func (n *network) createLoadBalancerSandbox() error {
 		}
 	}()
 
-	if err := ep.Join(sb, nil); err != nil {
+	err = ep.Join(sb, nil)
+	if err != nil {
 		return err
 	}
-	return sb.EnableService()
+
+	err = sb.EnableService()
```

Comment:
Very deliberately no.   This was the error that the patch needed to fix.  If one just does `return sb.EnableService()` then if `EnableService` returns an error that value does not get propagated to the function's `err` variable.   This, in turn, means that the cleanup defer functions will not detect the error and will thus not clean up after the error.

Reply:
ok then you can do this :) 
```
func (n *network) createLoadBalancerSandbox() (err error) {
```
This would keep the code clean. And avoid things like >>>>>>here

Reply:
That gets rid of the first line (err declaration for the function), but that's about it.  I'm not sure how that avoids things like `ep, err := n.CreateEndpoint(...)`.   The `:=` is (presumably) in the code because we don't want to declare the type of `ep`.  It's not to declare `err`.   Of course that has its own dangers.  If that statement were inside a loop, if-statement, switch statement or function body it would shadow the outer declaration causing disaster as well.   But that's true for named return values in the same way.  So I'm not sure what you're getting at.  Are you saying it should be changed to:
```
   var ep Endpoint
   ep, err = n.CreateEndpoint(...)
   ...
```

As far as the named return values and use of bare returns -- *shrug* it could be argued that this is less clear:
```
    x, err := foo()
    if err {
        return     // Looks at first glance like we aren't returning the error
    } 
```
and some prominent golang folks have argued that these should be used sparingly.  But I can put it in if desired.   The real issue is that the shadowing rules in golang are a bit of a misfeature.   One can see what they were aiming for, but there really aren't any clean-looking-while-still-robust patterns w/ regard to error handling because of them.  (although I'd be happy to learn otherwise)

Reply:
you can always do ```return err``` with the suggested syntax. It maps it to the defer value check and thats used in a lot of big golang code bases.

Reply:
```
func (n *network) createLoadBalancerSandbox() (retErr error) {
	sandboxName := n.lbSandboxName()
	sbOptions := []SandboxOption{}
	if n.ingress {
		sbOptions = append(sbOptions, OptionIngress())
	}
	sb, err := n.ctrlr.NewSandbox(sandboxName, sbOptions...)
	if err != nil {
		return err
	}
	defer func() {
		if retErr != nil {
			if e := n.ctrlr.SandboxDestroy(sandboxName); e != nil {
				logrus.Warnf("could not delete sandbox %s on failure on failure (%v): %v", sandboxName, err, e)
			}
		}
	}()

	endpointName := n.lbEndpointName()
	epOptions := []EndpointOption{
		CreateOptionIpam(n.loadBalancerIP, nil, nil, nil),
		CreateOptionLoadBalancer(),
	}
	if n.hasLoadBalancerEndpoint() && !n.ingress {
		// Mark LB endpoints as anonymous so they don't show up in DNS
		epOptions = append(epOptions, CreateOptionAnonymous())
	}
	ep, err := n.createEndpoint(endpointName, epOptions...)
	if err != nil {
		return err
	}
	defer func() {
		if retErr != nil {
			if e := ep.Delete(true); e != nil {
				logrus.Warnf("could not delete endpoint %s on failure on failure (%v): %v", endpointName, err, e)
			}
		}
	}()

	err := ep.Join(sb, nil)
	if err != nil {
		return err
	}

	return sb.EnableService()
}
```

Reply:
Ah, so a non-blank return statement assigns to the named return variable!  Perfect.  Much cleaner.  Will change.  Thanks!